www/caddy/src/opnsense/scripts/Pischem/Caddy/setup.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/sh
+
+# Define directories
+CADDY_DIR="/usr/local/etc/caddy"
+CADDY_ACME_DIR="${CADDY_DIR}/acme"
+CADDY_CERTS_DIR="${CADDY_DIR}/certificates/temp"
+CADDY_OCSP_DIR="${CADDY_DIR}/ocsp"
+CADDY_LOCKS_DIR="${CADDY_DIR}/locks"
+CADDY_LOG_DIR="/var/log/caddy/access"
+CADDY_CONF_DIR="${CADDY_DIR}/caddy.d"
+CADDY_CONFIG_DIR="${CADDY_DIR}/.config/caddy" # Additional config directory
+
+# Create Caddy configuration directories with appropriate permissions
+mkdir -p "${CADDY_DIR}"
+mkdir -p "${CADDY_ACME_DIR}"
+mkdir -p "${CADDY_CERTS_DIR}"
+mkdir -p "${CADDY_OCSP_DIR}"
+mkdir -p "${CADDY_LOCKS_DIR}"
+mkdir -p "${CADDY_CONF_DIR}"
+mkdir -p "${CADDY_CONFIG_DIR}"
+
+# Set permissions for Caddy configuration directories
+chown -R root:wheel "${CADDY_DIR}"
+chmod -R 750 "${CADDY_DIR}"
+
+# Create Caddy log directory
+mkdir -p "${CADDY_LOG_DIR}"
+
+# Set permissions for Caddy log directory
+chown -R root:wheel "${CADDY_LOG_DIR}"
+chmod -R 750 "${CADDY_LOG_DIR}"
+
+# Ensure the Caddy service script is executable
+chmod +x /usr/local/etc/rc.d/caddy
+
+# Ensure the Caddy binary is executable
+chmod +x /usr/local/bin/caddy
+
+# Format and overwrite the Caddyfile
+cd "${CADDY_DIR}" && /usr/local/bin/caddy fmt --overwrite
```

Comment:
```suggestion
(cd "${CADDY_DIR}" && /usr/local/bin/caddy fmt --overwrite)
```

a little subshell trick to avoid changing the directory of the currently executing script